stm/stm.ml

```diff
@@ -1794,6 +1981,61 @@ let log_processing_sync id name reason = log_string Printf.(sprintf
 let wall_clock_last_fork = ref 0.0
 
 let known_state ?(redefine_qed=false) ~cache id =
+ 
+  let error_absorbing_tactic id exn =
+    let boxes = VCS.box_of id in
+    let valid_boxes = CList.map_filter (function
+      | ErrorBound ({ stop } as decl, name) when Stateid.equal stop id ->
+          Some (decl, name)
+      | _ -> None) boxes in
+    assert(List.length valid_boxes < 2);
```

Comment:
Put anomaly here. (each tactic can be the error absorber for 1 and only 1 box).
Also, the static detection of proof blocks could catch case this early an report the error precisely.